src/intersections/lines.jl

```diff
@@ -43,14 +28,52 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
+
+ """
+    intersectparameters(a, b, c, d)
+
+Compute the parameters `λ₁` and `λ₂` of the lines 
+`a + λ₁ ⋅ v⃗₁`, with ` v⃗₁ = b - a` and
+`c + λ₂ ⋅ v⃗₂`, with ` v⃗₂ = d - c`
+spanned by the input points `a`, `b` resp. `c`, `d`
+such that to yield line points with minimal distance 
+or the intersection point (if lines intersect).
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+Furthermore the ranks `r` of the matrix of the linear system 
+`A ⋅ λ⃗ = y⃗`, with `A = [v⃗₁ -v⃗₂], y⃗ = c - a`
+and the rank `rₐ` of the augmented matrix `[A y⃗]` are calculated in order to 
+identify the intersection type:
 
-  # the intersection point lies in between a and b at a fraction s
-  # (https://en.wikipedia.org/wiki/Line-line_intersection#Formulas)
-  s = (v12[1] * v2[2] - v12[2] * v2[1]) / (v1[1] * v2[2] - v1[2] * v2[1])
+- Intersection: r == rₐ == 2
+- Collinear: r == rₐ == 1
+- No intersection: r != rₐ
+  - No intersection and parallel:  r == 1, rₐ == 2
+  - No intersection, skew lines: r == 2, rₐ == 3
+"""
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, 
+                             c::Point{N,T}, d::Point{N,T}) where {N,T}
+  A = [(b - a) (c - d)]
+  y = c - a
+  QRₐ = qr([A y])
+
+  # calculate the rank of the augmented matrix
+  # by checking the zero entries of the diagonal of R
+  # for N == 2 one has to check the L1 norm of rows as 
+  # there are more columns than rows
+  rₐ = N == 2 ? sum(sum(abs, QRₐ.R; dims = 2) .> atol(T)) :
+                sum(abs.(diag(QRₐ.R)) .> atol(T))
+  # calculate the rank of the rectangular matrix
+  r = sum(sum(abs, QRₐ.R[:,1:2]; dims = 2) .> atol(T))
+
+  if r ≥ 2 # calculate parameters of intersection or closest point
+    QR = qr(A)
+    λ = QR.R \ QR.Q' * y
```

Comment:
can't this expression be simplified to `QR \ y`?

Reply:
No that does not work, since QR is a static array (since A is a static Array).
If A were a normal array,  QR would be a so-called [QR object](https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.QR) that offers that direct feature